app/src/features/skjema-moduler/Inntekt.tsx

```diff
@@ -237,77 +233,202 @@ const EndreMånedslønn = ({ onClose, opplysninger }: EndreMånedslønnProps) =>
   };
 
-  const årsak = watch("inntektEndringsÅrsak.årsak");
   const inntekt = watch("inntekt");
 
   return (
-    <div>
+    <>
       <div className="flex items-start gap-4">
         <TextField
-          {...register("inntektEndringsÅrsak.korrigertInntekt", {
+          {...register("korrigertInntekt", {
             min: { value: 1, message: "Må være mer enn 0" },
             required: "Må oppgis",
             value: inntekt,
-            shouldUnregister: true,
           })}
-          error={
-            formState.errors.inntektEndringsÅrsak?.korrigertInntekt?.message
-          }
+          error={formState.errors.korrigertInntekt?.message}
           inputMode="numeric"
           label={`Månedsinntekt ${formatDatoKort(new Date(startdatoPermisjon))}`}
         />
-
-        <Select
-          className="flex-1"
-          error={formState.errors.inntektEndringsÅrsak?.årsak?.message}
-          label="Velg endringsårsak"
-          {...register("inntektEndringsÅrsak.årsak", {
-            required: "Må oppgis",
-            shouldUnregister: true,
-          })}
-        >
-          <option value="">Velg endringsårsak</option>
-          {Object.values(endringsårsak).map((årsak) => (
-            <option key={årsak.value} value={årsak.value}>
-              {årsak.label}
-            </option>
-          ))}
-        </Select>
         <Button
           className="mt-8"
           icon={<ArrowUndoIcon aria-hidden />}
           onClick={tilbakestillOgLukk}
+          type="button"
           variant="tertiary"
         >
           Tilbakestill
         </Button>
       </div>
-      <div className="flex gap-4 mt-4">
-        {NØDVENDIGE_FELTER_FOR_ÅRSAK[årsak]?.fom ? (
-          <DatePickerWrapped
-            label="Fra og med"
-            name="inntektEndringsÅrsak.fom"
-            rules={{ required: "Må oppgis" }}
-          />
-        ) : undefined}
-        {NØDVENDIGE_FELTER_FOR_ÅRSAK[årsak]?.tom ? (
-          <DatePickerWrapped
-            label="Til og med"
-            name="inntektEndringsÅrsak.tom"
-            rules={{ required: "Må oppgis" }}
-          />
-        ) : undefined}
-      </div>
-    </div>
+      <EndringsÅrsaker />
+    </>
   );
 };
 
+export const ENDRINGS_ÅRSAK_TEMPLATE = {
+  fom: undefined,
+  tom: undefined,
+  bleKjentFom: undefined,
+  årsak: "" as const,
+};
+
+function EndringsÅrsaker() {
+  const { eksisterendeInntektsmeldinger } = useLoaderData({ from: "/$id" });
+
+  const { control, register, formState } =
+    useFormContext<InntektOgRefusjonForm>();
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: "endringAvInntektÅrsaker",
+  });
+
+  // Tariffendring skal kun være tilgjengelig dersom man endrer en IM, ikke for førstegangs-innsendelse
+  const muligeÅrsakerValg =
+    eksisterendeInntektsmeldinger.length > 0
+      ? Object.values(endringsårsak)
+      : Object.values(endringsårsak).filter(
+          (årsak) => årsak.value !== "TARIFFENDRING",
```

Comment:
Har vi denne som en enum et sted? Trenger jo ikke kanskje, men digg å unngå streng-konstanter der det går an.

Reply:
Hm, er litt 🤷 either way for meg. Vi har jo typesikkerheten og vil få autocomplete også når vi skriver verdiene som streng.

Litt mer neat enn å måtte skrive `EndringAvInntektÅrsakerSchema.Enum.TARIFFENDRING`.